### ISSUE TITLE
Uses old Google authentication API

### ISSUE THREAD

**Issue by SamStephens** [splitbrain/googlephonefix#8]
The version of googlephonefix at http://splitbrain.github.com/googlephonefix/ shows the following message when you log in.

> This website is using an older Google authentication API that is no longer supported. 
> On July 2, 2012 it will stop working. 
> We recommend that you continue the process only if you trust the following destination:
> 
> ```
> http://splitbrain.github.com/googlephonefix/
> ```

**Comment by yoav-zibin** [comment]
Yes, now it no longer works completely

**Comment by nairbd** [comment]
Any plans to update the authentication? Really love this tool...

**Comment by splitbrain** [comment]
I haven't looked at the new API, yet. If it is possible I will upgrade to the new API but can not make any promises on when I'll find the time. Patches would be welcome.

**Comment by vonpupp** [comment]
Herewith another request. I will try to contribute, if I get it to work, I will send you the patch... =)

**Comment by splitbrain** [comment]
I just tried to fix this but failed. It seems that Google is phasing out the old gdata API JavaScript library in favor of their new gapi library. The latter works with OAuth2 and is really simple to use. Unfortunately Contacts aren't supported there. For the old gdata API the authsub authentication is no longer supported and I have absolutely no idea how to use OAuth2 with that library (I doubt it works). So as far as I understand, it is currently not possible to build a JavaScript only  application that can access Google Contacts.

I'd be happy to be taught otherwise.

**Comment by jcsalomon** [comment]
Is Google’s [Contacts API v3](https://developers.google.com/google-apps/contacts/v3/) at all relevant to this effort?

**Comment by zenware** [comment]
@jcsalomon Yes it is, looks like you can use OAuth2.0 and access the contacts that way, it's going to require people to create an app in the google developer console and enable the contacts API, as well as configure the scripts to use their ClientID and Secret. Also it requires some modifications to googlephonefix but I'm working on a patch.
